src/main/java/hudson/remoting/Channel.java

```diff
@@ -1314,10 +1316,12 @@ public synchronized void close(@CheckForNull Throwable diagnosis) throws IOExcep
      * (as opposed to executing {@link Callable}, which cannot have any reference to the context
      * of the remote {@link Channel}.
      */
-    public Object getProperty(Object key) {
+    @Override
+    public synchronized Object getProperty(Object key) {
```

Comment:
Hmmm are we sure we want to depend on the `Channel` monitor... for property access it may be preferential to use `properties` as the monitor object